src/test/java/org/fundacionjala/coding/dennis/TestCaseRental.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testMovieAndPrice() {
+        Customer c = new Customer("Xavier");
+        Rental rent = new Rental(new MovieRegular("Fargo"), DAYSRENTED4);
+        c.addRental(rent);
+        final float price = 6.0f;
+        String expected = rent.getMovie().getTitle().concat("\t\t").concat(String.valueOf(price));
+        assertEquals(expected, c.movieAndPrice(rent, price));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testCalculateAmountMovieChildrenPass() {
+        Movie mov = new MovieChildren("Tarzan");
+        final double res = 3.0;
+        assertEquals(res, mov.calculateAmount(DAYSRENTED4), LIMIT);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testReturnFrequentPointsMovieNewReleasePass() {
+        Movie mov = new MovieNewRelease("The house that Jack built");
+        final int expected = 2;
+        assertEquals(expected, mov.returnFrequentPoints(DAYSRENTED));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testReturnFrequentPointsMovieNewReleasePass1() {
+        Movie mov = new MovieNewRelease("The house that Jack built");
+        final int expected = 1;
+        assertEquals(expected, mov.returnFrequentPoints(1));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testCalculateAmountMovieNewReleasePass() {
+        Movie mov = new MovieNewRelease("A star is born");
+        final double res = 12.0;
+        assertEquals(res, mov.calculateAmount(DAYSRENTED4), LIMIT);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testCalculateAmountMovieRegularPass() {
+        final int daysRented = 5;
+        Movie mov = new MovieRegular("Sing Street");
+        final float res = 6.5f;
+        assertEquals(res, mov.calculateAmount(daysRented), LIMIT);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testGetDaysRentedRentalPass() {
+        Rental rent = new Rental(new MovieChildren("Inside Out"), DAYSRENTED);
+        final int expected = 3;
+        assertEquals(expected, rent.getDaysRented());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testGetMovieRentalPass() {
+        Rental rent = new Rental(new MovieNewRelease("Incredibles 2"), DAYSRENTED);
+        String expected = "Incredibles 2";
+        assertEquals(expected, rent.getMovie().getTitle());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testFrequentPoints() {
+        Customer customer = new Customer("Ramon");
+        customer.addRental(new Rental(new MovieNewRelease("Everybody knows"), 2));
+        customer.addRental(new Rental(new MovieRegular("Black Swan"), 2));
+        customer.addRental(new Rental(new MovieChildren("How to train your dragon"), 2));
+        customer.addRental(new Rental(new MovieNewRelease("Roma"), DAYSRENTED));
+        assertEquals(FPOINTS, customer.frequentPoints());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testAmount() {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "testAmount" to prevent any misunderstanding/clash with field "TESTAMOUNT" defined on line 16. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1845)